trunk/source/BoogiePreprocessor/src/de/uni_freiburg/informatik/ultimate/boogie/preprocessor/memoryslicer/MemoryArrayReplacer.java

```diff
@@ -239,6 +239,15 @@ protected Statement processStatement(final Statement statement) {
 		return super.processStatement(statement);
 	}
 
+	@Override
+	protected Expression processExpression(final Expression expr) {
+		if (expr instanceof final IdentifierExpression id && id.getIdentifier().startsWith("#memory")) {
```

Comment:
Nitpick: Use the constant `SFO.MEMORY` instead of a string literal.

Reply:
`SFO` is part of `CACSL2BoogieTranslator` which should not be imported by `BoogirPreprocessor`. There are more cases like that in `MemorySlicer` and this is a bit of a conceptional issue (but it does work 🙂).